modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/ExtensionModelHelper.java

```diff
@@ -382,14 +383,30 @@ protected void onNestable(ComposableModel owner, NestableElementModel model) {
               }
             }
 
+            @Override
+            protected void onType(MetadataType type) {
+              final Optional<DslElementSyntax> dslElement = resolveDslElementModel(type, currentExtension);
```

Comment:
what if the same type is found several times?

Reply:
will handle this in the walker itself